src/main/resources/public/js/getAndAddContracts.js

```diff
@@ -20,165 +20,147 @@ var projectContractsCount = -1;
+                                                if (activeWallet.type == 'FAKE') {
+                                                    message += ' You are using a fake wallet, the payment will be fictive.'
+                                                }
+                                                confirmDialog
+                                                    .create(message)
+                                                    .then(
+                                                        () => payInvoice(invoice, contract, $(this))
+                                                    );
                                             }
-                                            confirmDialog
-                                                .create(message)
-                                                .then(
-                                                    () => payInvoice(invoice, contract, $(this))
-                                                );
-                                        }
-                                    );
+                                        );
+                                    }
                                 }
                             }
-                        );
-                        $("#invoicesTable").dataTable();
-                        $("#invoicesBody").show();
-                    },
-                    204: function (data) {
-                        $("#loadingInvoices").hide();
-                        $("#invoicesTable").dataTable();
-                        $("#invoicesBody").show();
-                    },
-
-                }
+                        }
+                    }
+                );
             }
-        );
+        });
     }
 
     /**
-     * Turn an Invoice into a table row.
-     * @param invoice Invoice.
+     * Turn an Contract Invoice into a table row.
+     * @param contract Contract.
+     * @returns Function which will return and array of columns.
      */
-    function invoiceAsTableRow(contract, invoice) {
-        var status;
-        var payIcon = "";
-        var pdfHref = "/api/projects/"
-            + contract.id.repoFullName
-            + "/contracts/" + contract.id.contributorUsername + "/invoices/"
-            + invoice.id
-            + "/pdf?role=" + contract.id.role;
-        var downloadLink = "<a href='"+pdfHref+"' title='Download Invoice' class='downloadInvoice'>"
-            + "<i class='fa fa-file-pdf-o fa-lg'></i>"
-            + "</a>  ";
-        if (invoice.paymentTime == "null" && invoice.transactionId == "null") {
-            status = "Active";
-            if (parseFloat(invoice.totalAmount) > 0.0) {
-                payIcon = "<a href='#' title='Pay Invoice' class='payInvoice'>"
-                    + "<i class='fa fa-credit-card fa-lg'></i>"
-                    + "</a>";
+    function invoiceAsTableRow(contract) {
```

Comment:
@amihaiemil Same, the presentation can be done on the server.
```javascript
{
 id: 1,
 createdAt: "formatted date",
 amount: "formatted euro",
 status: "Paid|Active",
 options: {
   downloadLink : "<endpoint link>",
   paymentLink: "<endpoint link>"
 }
}
```
For options when rendering, we  only need to check if download/payment link is empty. If they are not empty then we attach the click listeners and showing icons, as usual.
Download/Payment link will be empty if amount is less than 108 euro, so we will avoid checking this on frontend.